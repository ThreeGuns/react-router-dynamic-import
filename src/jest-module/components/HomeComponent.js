/**
 * Create by cl on 2018/6/22
 */

import React from 'react';

class HomeComponent extends React.Component{
    render(){
        const {cnt, clickBtn, clickBtn2} = this.props;
        return (
            <div>
                <button id="btn-add" onClick={clickBtn}>click me to add</button>
                <button id="btn-reduce" onClick={clickBtn2}>click me to reduce</button>
                <label id={`cnt`}>{cnt}</label>
            </div>
        );
    }
}

export default HomeComponent;