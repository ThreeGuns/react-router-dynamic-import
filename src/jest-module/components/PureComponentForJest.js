/**
 * Create by cl on 2018/6/22
 */

import React from 'react';

class PureComponentForJest extends React.Component{
    render(){
        return (
            <div>
                this is pure component for jest
            </div>
        );
    }
}

export default PureComponentForJest;