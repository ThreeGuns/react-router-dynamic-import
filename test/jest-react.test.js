/**
 * Create by cl on 2018/6/22
 */

// import JestModule from '../src/jest-module';

import React from 'react';
import PureComponentForJest from '../src/jest-module/components/PureComponentForJest';
import JestModuleRoot from '../src/jest-module/containers/Root';
import HomeComponent from '../src/jest-module/components/HomeComponent';
import renderer from 'react-test-renderer';

test('jest-module test', () => {
    const component = renderer.create(<PureComponentForJest/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const jestModuleRoot = renderer.create(<JestModuleRoot/>);
    let tree2 = jestModuleRoot.toJSON();
    expect(tree2).toMatchSnapshot();

    const homeComponent = renderer.create(<HomeComponent/>);
    let tree3 = homeComponent.toJSON();
    //console.log(tree3);
    // homeComponent.props.clickBtn();
});