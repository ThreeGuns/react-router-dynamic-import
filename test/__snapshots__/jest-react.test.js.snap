// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-module test 1`] = `
<div>
  this is pure component for jest
</div>
`;

exports[`jest-module test 2`] = `
<div
  className="jest-module"
>
  <div>
    <button
      id="btn-add"
      onClick={[Function]}
    >
      click me to add
    </button>
    <button
      id="btn-reduce"
      onClick={[Function]}
    >
      click me to reduce
    </button>
    <label
      id="cnt"
    >
      0
    </label>
  </div>
  <div>
    <a
      id="footer-a"
      onClick={[Function]}
    >
      click footer to back
    </a>
  </div>
</div>
`;
